File: geqiangbao_20201117_CommandPattern_01/geqiangbao_20201117_CommandPattern_01/day1_Command/computer/MacStartUpCommand.m
//
//  MacStartUpCommand.m
//  geqiangbao_20201117_CommandPattern_01
//
//  Created by xiaobao on 2020/11/17.
//

#import "MacStartUpCommand.h"
@interface MacStartUpCommand ()
@property (nonatomic,strong) MacComputer *computer;
@end

@implementation MacStartUpCommand
-(instancetype)initWith:(MacComputer*)computer{
    self = [super init];
    if (self) {
        self.computer = computer;
    }
    return self;
}



-(void)execute{
    [self.computer startUp];
}

@end

File: geqiangbao_20201117_CommandPattern_swift/geqiangbao_20201117_CommandPattern_swift/computer/MacStartUpCommand.swift
//
//  MacStartUpCommand.swift
//  geqiangbao_20201117_CommandPattern_swift
//
//  Created by xiaobao on 2020/11/17.
//

import Cocoa

class MacStartUpCommand: NSObject,MacComputerProtocol {

    private var computer:MacComputer
    init(_ computer:MacComputer) {
        self.computer = computer;
    }
    func execute() {
        self.computer.startUp()
    }
}
